src/Interpreters/InterpreterSystemQuery.cpp

```diff
@@ -400,6 +403,96 @@ void InterpreterSystemQuery::restartReplicas(Context & system_context)
     pool.wait();
 }
 
+void InterpreterSystemQuery::dropReplicaByZKPath(ASTSystemQuery & query)
+{
+    context.checkAccess(AccessType::SYSTEM_DROP_REPLICA);
+    auto to_drop_path = query.replica_zk_path + "/replicas/" + query.replica;
+    auto & catalog = DatabaseCatalog::instance();
+    StorageReplicatedMergeTree::Status status;
+
+    for (auto & elem : catalog.getDatabases())
+    {
+        DatabasePtr & database = elem.second;
+        for (auto iterator = database->getTablesIterator(context); iterator->isValid(); iterator->next())
+        {
+            if (auto * storage_replicated = dynamic_cast<StorageReplicatedMergeTree *>(iterator->table().get()))
+            {
+                storage_replicated->getStatus(status);
+
+                if (status.replica_path.compare(query.replica_zk_path + "/replicas/" + status.replica_name) == 0)
+                {
+                    storage_replicated->dropReplica(query.replica, false);
+                    return;
+                }
+            }
+        }
+    }
+    auto remote_replica_path = query.replica_zk_path  + "/replicas" + "/" + query.replica;
+    LOG_INFO(log, "Removing replica {}", remote_replica_path);
+    /// It may left some garbage if replica_path subtree are concurently modified
+    auto zookeeper = context.getZooKeeper();
+    zookeeper->tryRemoveRecursive(remote_replica_path);
```

Comment:
We should check that replica is not active before removing it. Also we probably should remove it the same way as `StorageReplicatedMergeTree::dropReplica(...)` does. It can be done by extracting some part of `StorageReplicatedMergeTree::dropReplica(...)` to static method.